### ISSUE TITLE
KubernetesPodOperator in deferrable mode does not delete pod when task is marked externally

### ISSUE THREAD

**Issue by kien-truong** [apache/airflow#44014]
### Apache Airflow version

2.10.3

### What happened?

KubernetesPodOperator in deferrable mode does not delete pod when task is marked fail/success manually in the UI.

### What you think should happen instead?

KubernetesPodOperator in deferrable mode should delete pod when the task is marked fail/success

### How to reproduce

KubernetesPodOperator in deferrable mode does not delete pod when task is marked fail

### Operating System

Linux

### Versions of Apache Airflow Providers

apache-airflow-providers-cncf-kubernetes==9.0.0

### Deployment

Google Cloud Composer

### Deployment details

Composer 2.9.9
Airflow 2.9.3

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by Lee-W** [comment]
Yep, I think we'll probably need to wait and see how https://github.com/apache/airflow/issues/36090 is handled
